app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -731,6 +766,74 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Activit
         initializeEngineView(toolbarHeight)
     }
 
+    /**
+     * Shows a biometric prompt and fallback to prompting for the password.
+     */
+    private fun showBiometricPrompt(context: Context) {
+        if (BiometricPromptFeature.canUseFeature(context)) {
+            biometricPromptFeature.get()
+                ?.requestAuthentication(getString(R.string.credit_cards_biometric_prompt_unlock_message))
+            return
+        }
+
+        // Fallback to prompting for password with the KeyguardManager
+        val manager = context.getSystemService<KeyguardManager>()
+        if (manager?.isKeyguardSecure == true) {
+            showPinVerification(manager)
+        } else {
+            // Warn that the device has not been secured
+            if (context.settings().shouldShowSecurityPinWarning) {
+                showPinDialogWarning(context)
+            } else {
+                promptsFeature.get()!!.onActivityResult(
+                    requestCode = PIN_REQUEST,
+                    data = null,
+                    resultCode = Activity.RESULT_OK
+                )
+            }
+        }
+    }
+
+    /**
+     * Shows a pin request prompt. This is only used when BiometricPrompt is unavailable.
+     */
+    @Suppress("Deprecation")
+    private fun showPinVerification(manager: KeyguardManager) {
+        val intent = manager.createConfirmDeviceCredentialIntent(
+            getString(R.string.credit_cards_biometric_prompt_message_pin),
+            getString(R.string.credit_cards_biometric_prompt_unlock_message)
+        )
+        startActivityForResult(intent, PIN_REQUEST)
+    }
+
+    /**
+     * Shows a dialog warning about setting up a device lock PIN.
+     */
+    private fun showPinDialogWarning(context: Context) {
+        AlertDialog.Builder(context).apply {
+            setTitle(getString(R.string.credit_cards_warning_dialog_title))
+            setMessage(getString(R.string.credit_cards_warning_dialog_message))
+
+            setNegativeButton(getString(R.string.credit_cards_warning_dialog_later)) { _: DialogInterface, _ ->
+                promptsFeature.get()!!.onActivityResult(
+                    requestCode = PIN_REQUEST,
+                    data = null,
+                    resultCode = Activity.RESULT_CANCELED
+                )
```

Comment:
It think we should use instead `promptsFeature.get()?.onBiometricResult( isAuthenticated = false)`